README.md

```diff
@@ -105,6 +105,41 @@ Use `#pragma mark -` to categorize methods in functional groupings and protocol/
 - (NSString *)description {}
 ```
 
+## Header Imports
+
+#import statements should be kept to a minimum, especially in header files. Within header files use @class and @protocol where possible to avoid importing those dependencies.
+
+Headers should at least, directly or indirectly, import <Foundation/Foundation.h> (any common SDK framework header will import foundation indirectly).
+
+Framework imports should always import the framework umbrella header, not any of it individual header files.
```

Comment:
@Gabardone what downsides do you see when only required headers of a framework are imported?

Reply:
Public framework headers are often not really designed for being individually imported and might behave in somewhat unexpected ways (especially if there's a lot of macros involved), and just hitting the umbrella avoids tightly coupling the import with the specific header where the symbols needed are declared. Which helps separate modules and makes for easier maintenance on both sides.

Reply:
Thank you for the answer.